### PR TITLE
Fixing the csproj generation test

### LINKED ISSUE #981: Worker sdk requires dotnet 7 and dotnet 6 to build
Currently a customer needs to install dotnet 6 and dotnet 7, if they are targeting net7. We need to update the internal csproj generation to use the same TFM as the one from the functions project. This would make it so that the customer only needs to install dotnet 6 or dotnet 7, but not both.

### PR DESCRIPTION
**Pull request by soninaren**
### Issue describing the changes in this PR

Fixing the csproj generation test

resolves #981

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by liliankasem — APPROVED** [review]

**Comment by soninaren** [comment]
including the change in https://github.com/Azure/azure-functions-dotnet-worker/pull/1014, as the build is currently broken with both changes

File: src/DotNetWorker.Core/Context/FunctionContextHttpRequestExtensions.cs [hunk 1]
```diff
@@ -45,7 +47,7 @@ public static class FunctionContextHttpRequestExtensions
             }
 
             // see output binding entries have a property of type HttpResponseData;
-            var httpOutputBinding = context.GetOutputBindings<HttpResponseData>().FirstOrDefault();
+            var httpOutputBinding = context.GetOutputBindings<HttpResponseData>().First(a => a.BindingType == HttpBindingType);
```
Comment (kshyju) [inline src/DotNetWorker.Core/Context/FunctionContextHttpRequestExtensions.cs:50]: Let's change to FirstOrDefault

Comment (soninaren) [reply src/DotNetWorker.Core/Context/FunctionContextHttpRequestExtensions.cs]: done

**Pull request merged**